subprojects/docs/src/docs/userguide/javaTesting.adoc

```diff
@@ -377,16 +419,19 @@ This will set up a new source set called `intTest` that automatically creates:
  * `intTestImplementation`, `intTestCompileOnly`, `intTestRuntimeOnly` configurations (and <<java_source_set_configurations, a few others>> that are less commonly needed)
  * A `compileIntTestJava` task that will compile all the source files under _src/intTest/java_
 
-The example also does the following:
+The example also does the following, not all of which you may need for your specific integration tests:
 
  * Adds the production classes from the `main` source set to the compilation and runtime classpaths of the integration tests — `sourceSets.main.output` is a <<sec:file_collections,file collection>> of all the directories containing compiled production classes and resources
  * Makes the `intTestImplementation` configuration extend from `implementation`, which means that all the declared dependencies of the production code also become dependencies of the integration tests
+ * Does the same for the `intTestRuntimeOnly` configuration
 
-Another common step is to attach all the unit test dependencies to the integration tests as well — via `intTestImplementation.extendsFrom testImplementation` — but that only makes sense if the integration tests require all the dependencies that the unit tests have.
+In most cases, you want your integration tests to have access to the classes under test, which is why we ensure that those are included on the compilation and runtime classpaths. But certain types of test interact with an application as an executable or via HTTP for web applications. There is no need in those cases to add the production classes to the tests' classpaths.
```

Comment:
❗️ This feels a little off: "no need in those cases to add"

Reply:
Reworked most of the paragraph. Hope it's easier to understand now.